lua/cmp/source.lua

```diff
@@ -128,16 +127,17 @@ source.get_entries = function(self, ctx)
 
       if entry_filter(e, ctx) then
         entries[#entries + 1] = e
-        if max_item_count and #entries >= max_item_count then
-          break
-        end
       end
     end
+    async.yield()
+    if ctx.aborted then
+      async.abort()
+    end
   end
 
-  -- only save to cache, when there are no additional entries that could match the filter
-  -- This also prevents too much memory usage
-  if #entries < max_item_count then
+  -- only save to cache, when there are not too many entries
+  -- This prevents too much memory usage
+  if #entries < 200 then
     self.cache:set({ 'get_entries', tostring(self.revision), ctx.cursor_before_line }, entries)
   end
```

Comment:
I'll change to be the latest only full cache later.

Reply:
What do you mean exactly? I don't fully understand your comment :)

Reply:
Currently, nvim-cmp will create cache for each user input.

1. user-input=`use`, cache-key: `get_entries:%REVISION%:use`
2. user-input=`useE`, cache-key: `get_entries:%REVISION%:useE`
3. ...

This has benefit for performance when user press the backspace key.

But now folke research shows that this cache is bad for memory.

It doesn't make sense to cache or not by the number of items here.

Either keep a cache only for the last user input, or remove caching entirely.

Reply:
I think I correctly implemented this in https://github.com/hrsh7th/nvim-cmp/pull/1583/commits/c0bf17c4b228e7a24975675425c15b72c06a876d

Would be great if you could double-check if the re-use conditions are 100% correct!

I also removed the `filtered.entries` from my previous PR, since that is now no longer needed.